### PR TITLE
[Snaps] Add tx-insights test for V2

### LINKED ISSUE #21763: [FLASK] Create E2E test for Transaction Insights v2

### PR DESCRIPTION
**Pull request by bowensanders**
## **Description**

This PR enhances tx-insights test for V2, including a new warning and confirmation screen.

## **Related issues**

Fixes: #21763 

## **Manual testing steps**

1. CI passing means the test works

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by FrederikBolding — COMMENTED** [review]
I think we should have a test for v1 and one for v2. What do you think?

**Comment by bowensanders** [comment]
Agreed, moved to seperate test and added it flask only in run-all.js!

File: test/e2e/snaps/test-snap-txinsights-v2.spec.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        await driver.clickElement({
+          text: 'Confirm',
+          tag: 'button',
+        });
+
+        // check for warning from txinsights
+        await driver.waitForSelector({
+          css: '.snap-delineator__header__text',
+          text: 'Warning from Insights Example Snap',
+        });
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-txinsights-v2.spec.js:124]: Can we verify that the snaps custom UI is rendered in the warning too?

File: test/e2e/snaps/test-snap-txinsights-v2.spec.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+        // click the warning confirm checkbox
+        await driver.clickElement('.mm-checkbox__input');
+
+        // click confirm button to send transaction
+        await driver.clickElement({
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-txinsights-v2.spec.js:130]: Can we verify that the tx is added to the activity screen?

File: test/e2e/snaps/test-snap-txinsights-v2.spec.js (hunk 1 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-txinsights-v2.spec.js:124]: fixed in latest commit

File: test/e2e/snaps/test-snap-txinsights-v2.spec.js (hunk 2 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-txinsights-v2.spec.js:130]: added in latest commit

**Review by FrederikBolding — DISMISSED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**